microcosm_postgres/types.py

```diff
@@ -36,3 +36,44 @@ def process_result_value(self, value, dialect):
         if value is None:
             return None
         return self.enum_class[value]
+
+
+class Serial(UserDefinedType):
+    """
+    A postgres Serial type.
+
+    Intended for use with auto-incrementing immuatable columns that are NOT primary keys.
+
+    Use in conjuction with `server_default` to ensure that SQLAlchemy fetches the generated value.
+
+        mycolumn = Column(Serial, server_default=FetchedValue(), nullable=False)
```

Comment:
That's going to depend on the use case. In some cases, yes; in some cases we'll want multi-column constraints; in some cases maybe none.